include/config.js

```diff
@@ -113,585 +93,287 @@ var CONFIG_MODULE_NAME  = 'config.js';
  */
 var OVERRIDE_FILE_PATHS = [
     path.join(Configuration.DOCUMENT_ROOT, CONFIG_MODULE_NAME),
-    path.join(Configuration.EXTERNAL_ROOT, CONFIG_MODULE_NAME),
+    path.join(Configuration.EXTERNAL_ROOT, CONFIG_MODULE_NAME)
 ];
 
-/**
- * @private
- * @static
- * @readonly
- * @property BASE_CONFIG
+/** 
+ * Retrieve the base configuration
  */
-var BASE_CONFIG = {
-
-    //The name of the site.
-	siteName: 'pencilblue',
-
-    //The root of the site.  This host part should ALWAYS match the value of
-    //the siteIP
-	siteRoot: 'http://localhost:8080',
-
-    //The hostname or IP address that the web server instance will bind to
-	siteIP:   '0.0.0.0',
-
-    //The primary port to listen for traffic on.  Some environment such as
-    //heroku force you to use whatever port they have available.  In such cases
-    //the port is passed as an environment variable.
-	sitePort: process.env.port || process.env.PORT || 8080,
-
-    //the absolute file path to the directory where installation lives
-	docRoot:  Configuration.DOCUMENT_ROOT,
-
-    //provides a configuration for connecting to persistent storage.  The
-    //default configuration is meant for mongodb.
-	db: {
-        type:'mongo',
-		servers: [
-          '127.0.0.1:27017'
-        ],
-
-        //the name of the default DB for the system
-        name: 'pencil_blue',
-
-        //http://docs.mongodb.org/manual/core/write-concern/
-        writeConcern: 1,
-
-        //PB provides the ability to log queries.  This is handy during
-        //development to see how many trips to the DB a single request is
-        //making.  The queries log at level "info".
-        query_logging: false,
-
-        //http://mongodb.github.io/node-mongodb-native/api-generated/db.html#authenticate
-        authentication: {
-            un: null,
-            pw: null,
-            options: {
-                //authMechanism: "MONGODB-CR"|"GSSAPI"|"PLAIN", //Defaults to MONGODB-CR
-                //authdb: "db name here", //Defaults to the db attempted to be connected to
-                //authSource: "db name here", //Defaults to value of authdb
-            }
+Configuration.getBaseConfig = function(multisite) {
+    return util.clone({
```

Comment:
If you are creating the object anyway there is no need to clone it.  I would prefer that we move it back out to its own variable ```BASE_CONFIG``` as before.  It gives more room for comments before the print margin and prevents the actual structure of the base configuration from being tampered with if the ```Configuration.getBaseConfig``` function is overriden.